docs/tokencache.md

```diff
@@ -2,16 +2,350 @@
 
 ## Introduction
 
-> PENDING
+The token cache implementation for MSAL, ADAL and Android broker is found in the Android Common Library.  
 
-## Class Diagram
+## References 
 
-> PENDING
+- Universal Cache Schema
 
-## Interaction Diagram
+## Universal cache and protocol response handling
 
-> PENDING
+When looking at the token cache there are really 2 levels to it.  The lower level is based on the universal cache schema referenced above.  The higher level deals with the result of a particular protocol request and mapping that to the universal schema.  
+
+So in effect we get an oAuth 2 Token Response, which in the case of MSAL includes an ID Token and an Access Token and we translate:
```

Comment:
```suggestion
So in effect we get an oAuth2 Token Response, which in the case of MSAL includes an ID Token and an Access Token and we translate:
```

Reply:
Actually may be rather `OAuth 2.0`

This is just a nit...just trying myself to figure out the best way to write oauth2 :)

Reply:
I linked to the spec as well. ;-)